imports/server/hooks/ChatNotificationHooks.ts

```diff
@@ -71,7 +72,7 @@ const ChatNotificationHooks: Hookset = {
     await Promise.all(collected.map(async (userId: string) => {
       await ChatNotifications.insertAsync({
         user: userId,
-        sender: chatMessage.sender,
+        sender,
```

Comment:
For what it's worth, this is the result of slightly better type precision than we had previously, and I'm kind of pleased about it.

Because this is accessing an attribute in a callback, TypeScript assumes that the object could potentially be mutated, and thus ignores the truthiness check previously on line 22 - within this function, `chatMessage.sender` is of type `string | undefined`.

With the old `Base` logic, `insert` and `insertAsync` accepted a `Partial<T>`. We didn't have any way at the type layer to track whether a field would be auto-populated or not, but that meant that the type layer would accept missing or null fields that were required by the schema. (Validating the object against SimpleSchema would throw a runtime error, ultimately ensuring consistency)

However, the new `Model` implementation can tell the difference (by using Zod's "input" vs. "output" construct), and correctly flagged that, if `chatMessage.sender` is optional, then this is potentially invalid.